tests/executors/test_kubernetes_executor.py

```diff
@@ -1249,12 +1249,13 @@ def effect():
             try:
                 # self.watcher._run() is mocked and return "500" as last resource_version
                 self.watcher.run()
+                assert False, "Should have raised Exception"
```

Comment:
The change in line 1258 is all good to me.
But is this change at line 1252 necessary? I don't fully get it yet . The watcher is already mocked and will fail

Reply:
This is just consistency change.  The author actually expected the sentinel exception (and it's the one that is raised form the watcher to break the loop) but the path when (potentially) no exception is thrown would not run this assert:

```
except Exception as e:
                assert e.args == ("sentinel",)
```

Sure, we know what happens inside the run method and by inspecting it, we know it is not going to happen most likely - because exception is the only way to get out of the loop. But ... this might change in the future.

Not likely and stupid example but if someonedoes fast "return 0" in the `watcher.run()` - the test would have succeeded as well (no exception and resource = 0, `assert e.args == ("sentinel", )` would have not been called). 

Adding the assert simply makes absolutely sure that the exception was thrown (because otherwise assert would fail). That's the usual pattern :)